src/TwigComponent/src/ComponentPropertiesExtractor.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Twig\PropsNode;
+use Twig\Environment;
+
+final class ComponentPropertiesExtractor
+{
+    public function __construct(
+        private readonly Environment $twig,
+    ) {
+    }
+
+    /**
+     * @return array<string, string>
+     */
```

Comment:
If we create a class for this, we will need to be a little more precise about the types

Reply:
Thanks, that's fixed